sdk/go/auto/local_workspace.go

```diff
@@ -300,12 +300,18 @@ func (l *LocalWorkspace) PulumiVersion() string {
 }
 
 // WhoAmI returns the currently authenticated user
-func (l *LocalWorkspace) WhoAmI(ctx context.Context) (string, error) {
-	stdout, stderr, errCode, err := l.runPulumiCmdSync(ctx, "whoami")
+func (l *LocalWorkspace) WhoAmI(ctx context.Context) (WhoAmIResult, error) {
```

Comment:
Sounds good to me. Just a note: `WhoAmIDetailed` is definitely apt, perhaps `WhoAmIVerbose` would keep greater parity with CLI semantics. WDYT?

Reply:
Verbose is generally in terms of logging, and that's not the case here. I think I'd prefer Detailed vs Verbose, but I'll nudge internally just in case anyone knows some prior example of this case (adding a new Go method to return more information) where we've already got style.

Reply:
Good point. SGTM!